File: Examples/CSharp/AsposePdfFacades/Bookmarks/ExportToXML.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2014 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.Facades;

namespace CSharp.AsposePDFFacades.Bookmarks
{
    public class ExportToXML
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdfFacades_Bookmarks();
            //Create PdfBookmarkEditor object
            PdfBookmarkEditor bookmarkEditor = new PdfBookmarkEditor();
            //Open PDF file
            bookmarkEditor.BindPdf(dataDir+ "ExportToXML.pdf");
            //Export bookmarks
            bookmarkEditor.ExportBookmarksToXML(dataDir+ "bookmarks.xml");
            //Save updated PDF
            bookmarkEditor.Save(dataDir+ "ExportToXML_out.pdf");
            
            
        }
    }
}

File: Examples/VisualBasic/AsposePdfFacades/Bookmarks/ExportToXML.vb
Imports System.IO
Imports System
Imports Aspose.Pdf
Imports Aspose.Pdf.Facades

Namespace AsposePDFFacades.Bookmarks
    Public Class ExportToXML
        Public Shared Sub Run()
            ' ExStart:ExportToXML
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdfFacades_Bookmarks()
            ' Create PdfBookmarkEditor object
            Dim bookmarkEditor As New PdfBookmarkEditor()
            ' Open PDF file
            bookmarkEditor.BindPdf(dataDir & Convert.ToString("ExportToXML.pdf"))
            ' Export bookmarks
            bookmarkEditor.ExportBookmarksToXML(dataDir & Convert.ToString("bookmarks.xml"))
            ' Save updated PDF
            bookmarkEditor.Save(dataDir & Convert.ToString("ExportToXML_out_.pdf"))
            ' ExEnd:ExportToXML            
        End Sub
    End Class
End Namespace
